src/makeRobot.js

```diff
@@ -38,7 +38,68 @@
  * @return {Robot}
  */
 function makeRobot(name, wheels, version) {
-  // write code here
+  const robot = {
+
+    coords: {
+      x: 0,
+      y: 0,
+    },
+
+    get info() {
+      return `name: ${name}, chip version: ${version}, wheels: ${wheels}`;
+    },
+
+    get location() {
+      return `${name}: x=${this.coords.x}, y=${this.coords.y}`;
+    },
+
+    goBack(step = 1) {
+      if (step < 0) {
+        return this;
+      } else {
+        this.coords.y -= step;
+      }
```

Comment:
```suggestion
      if (step > 0) {
        this.coords.y -= step;
      }
```

It's best to avoid any steps that have a value of less than 1. Additionally, if you find yourself repeating code (such as using "return this;" multiple times), it's likely that there is room for optimization.